torchtune/modules/tokenizer.py

```diff
@@ -72,16 +79,25 @@ def encode(
             add_eos (bool): Whether to append EOS to the input, defaults to True.
             trim_leading_whitespace (bool): Whether to trim leading whitespace from
                 underlying sentencepiece tokenization. Default: False
+            prefix (Optional[str]): Optional string to encode for trimming leading
+                whitespaces. Default: None
         Returns:
             List[int]: The encoded token IDs.
         """
         if trim_leading_whitespace:
+            # Can define our own custom prefix depending on vocab if needed
+            if not hasattr(self, "prefix"):
```

Comment:
This is basically just caching it the first time we use it, hence why it's set in the following line. If it's already been encoded, no need to keep re-encoding it since it will remain fixed for the life of the program.